File: java/Crossword/Cell.java
public class Cell {
    private final boolean m_IsStartCell;
    private int m_WordsFromTheCell;
    private char m_CurrentChar;

    public Cell(int wordsFromTheCell) {
        m_IsStartCell = wordsFromTheCell > 0;
        m_WordsFromTheCell = wordsFromTheCell;
        m_CurrentChar = ' ';
    }

    public boolean isStartCell() {
        return m_IsStartCell;
    }

    public int wordsFromTheCell() {
        return m_WordsFromTheCell;
    }

    public char getCurrentChar() {
        return m_CurrentChar;
    }

    public void setCurrentChar(char c) {
        m_CurrentChar = c;
    }

    public void applyWordCountDelta(int delta) {
        m_WordsFromTheCell += delta;
    }

    @Override
    public String toString() {
        return Character.toString(m_CurrentChar);
    }
}

File: csharp/Crossword/Cell.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Crossword
{
    internal record struct Cell
    {
        public static readonly Cell None = new();

        public Cell(int wordsFromHere)
        {
            this.WordsFromHere = wordsFromHere;
        }

        public int WordsFromHere{get; init;}

        public char Character{get; init;} = ' ';

        public bool IsStartCell
        {
            get{return this.WordsFromHere > 0;}
        }

        public override string ToString()
        {
            if(this.Character == '\0') return ".";
            
            return this.Character.ToString();
        }

        public static Cell FromChar(char c)
        {
            if(c == '.') return Cell.None;

            var wordCount = c - '0';
            return new(wordCount);
        }
    }
}
